keps/sig-node/4112-passdown-resources-to-cri/README.md

```diff
@@ -0,0 +1,1277 @@
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Unit tests for the feature gate will be added.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+Rollback or rollout in the kubelet should not fail - it only enables/disabled
+the information (fields in the CRI message) passed down to the CRI runtime.
+
+However, if the CRI runtime depends on the feature, a rollout or rollback may
+cause failures of applications on pod restarts. Running pods are not affected.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Alpha: No new metrics are planned. Non-ready pods with CreatePodSandboxError
+status is one indicator. The error message will contain details if the CRI
+failure is related to the feature.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+Alpha: Manual testing of the feature gate is performed.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+By examing the kubelet configuration (feature gate) and the version of the
```

Comment:
@logicalhan added metrics for all feature gates, at least in API server. Han, is that in all components (kubelet in particular)?

@marquiz, if all that matters is whether the feature gate is on or off (which I think is true, it's why it's a default behavior change for that kubelet), then this metric is sufficient to determine if it is in use.

Reply:
Thanks for pointing this out. I modified this section,  proposing the `kubernetes_feature_enabled` metric to be used for checking kubelet enablement. PTAL @johnbelamaric

Reply:
Closing this as resolved